src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -413,13 +413,36 @@ protected void InitializeReferences()
             try
             {
                 runner.InitializeTimeout(internalMatchTimeout);
-                runner.InitializeForScan(this, input, startat, quick: true);
+                runner.InitializeForScan(this, input, startat, quick);
+
+                int stoppos = RightToLeft ? 0 : input.Length;
```

Comment:
This value is only used if prevlen == 0.  It should be moved to be inside of the if block below.  We can then also avoid extra branches around RightToLeft:
```C#
// If previous match was empty or failed, advance by one before matching.
if (prevlen == 0)
{
    if (RightToLeft)
    {
        if (runner.runtextstart == 0)
        {
            return RegularExpressions.Match.Empty;
        }
        runner.runtextpos--;
    }
    else
    {
        if (runner.runtextstart == input.Length)
        {
            return RegularExpressions.Match.Empty;
        }
        runner.runtextpos++;
    }
}
```